wallpaper.m

```diff
@@ -38,7 +38,37 @@ int main() {
 				return 1;
 			}
 		} else {
-			printf("%s\n", [sw desktopImageURLForScreen:screen].path.UTF8String);
+      NSString *url = [sw desktopImageURLForScreen:screen].path;
+      BOOL isDir;
+      // check if file is a directory
+      [[NSFileManager defaultManager] fileExistsAtPath:url isDirectory:&isDir];
+      
+      // if directory, check db
+      if (isDir) {
+        NSTask *task = [[NSTask alloc] init];
+
+        NSString *dbLoc = [NSHomeDirectory() stringByAppendingPathComponent:@"/Library/Application Support/Dock/desktoppicture.db"];
+
+        [task setLaunchPath: @"/usr/bin/sqlite3"];
```

Comment:
I think it would be better to use `sqlite3_open`, rather than spawning the process.

http://stackoverflow.com/questions/23559263/opening-and-creating-sqlite-database-in-objective-c